### PR TITLE
Fix mdx support

### LINKED ISSUE #66: Plugin does not work when configured as a gatsby-mdx gatsbyRemarkPlugin
Hi there! I'm trying to use this plugin with Gatsby and MDX, but I can't seem to make it work. I've got it listed in the `gatsby-mdx` `gatsbyRemarkPlugins` config, which has worked for other Remark plugins, but can't seem to find twitter links in my page. I'm not sure if where the problem is, but I don't see any errors. 

Here is my config for `gatsby-mdx`:

```js
	{
      resolve: `gatsby-mdx`,
      options: {
        defaultLayouts: {
          posts: require.resolve('./src/templates/blog-post.js'),
          default: require.resolve('./src/components/layout.js'),
        },
        extensions: ['.mdx', '.md'],
        gatsbyRemarkPlugins: [
          {
            resolve: '@raae/gatsby-remark-oembed',
            options: {
              providers: {
                settings: {
                  Twitter: {
                    theme: 'dark', // Use the Twitter dark theme
                  },
                },
              },
            },
          },
          {
            resolve: `gatsby-remark-copy-linked-files`,
          },
          {
            resolve: `gatsby-remark-relative-images`,
          },
          {
            resolve: `gatsby-remark-lazy-load`,
          },
          {
            resolve: `gatsby-remark-images`,
            options: {
              maxWidth: 768,
            },
          },
          {
            resolve: 'gatsby-remark-prismjs',
            options: {
              classPrefix: 'language-',
              inlineCodeMarker: {
                tsx: 'tsx',
              },
              aliases: {},
            },
          },
        ],
      },
    }
```

and here is the output from this plugin:

```
info gatsby-remark-oembed: Successfull embeds: 0 | Links with no matching provider: 6 | Path: /personal-site-hosting/
info gatsby-remark-oembed: Successfull embeds: 0 | Links with no matching provider: 1 | Path: /linked-lists-react/
info gatsby-remark-oembed: Successfull embeds: 0 | Links with no matching provider: 10 | Path: /accessibility-rock-star/
info gatsby-remark-oembed: Successfull embeds: 0 | Links with no matching provider: 2 | Path: /npm-card/
```

The post looks like this:

```md
I recently saw this awesome post from [bnb](https://dev.to/bnb) on Twitter:

https://twitter.com/bitandbang/status/1075473070368919552

The community response has been awesome and I quickly knew I wanted to make my own:

https://twitter.com/callmewuz/status/1075789700898664450
```

Thanks so much for making this plugin! I hope I can get it working 😄

### PR DESCRIPTION
**Pull request by Swizec**
This fixes #66 

The issue was that `gatsby-plugin-mdx` implements `onPreBootstrap`, which overrides oembed's onPreBootstrap implementation. This meant that when we tried to change embed nodes, we couldn't find any providers so all nodes got ignored.

I moved `onPreBootstrap` code to a separate helper and call it the same way as before. This means behavior when used with Remark should be unchanged.

However, when we try to use the list of providers, we now re-fetch it and update cache if it isn't yet available in cache. This makes the plugin work with MDX.

Here's an example repository with 1 post where using `gatsby-remark-oembed` worked with mdx: https://github.com/Swizec/swizec-blog/tree/oembed-playground

### CODE REVIEW

File: utils/getProviders.js [hunk 1]
```diff
@@ -0,0 +1,39 @@
+const amendOptions = require("./amendOptions");
+const amendProviders = require("./amendProviders");
+const filterProviders = require("./filterProviders");
+const fetchOembedProviders = require("./fetchOembedProviders");
```
Comment (Swizec) [inline utils/getProviders.js]: @raae this is a little silly. I couldn't for the life of me figure out why node wouldn't let me `const { ... } = require('./index')`.

Comment (raae) [reply utils/getProviders.js]: Yeah, I was very unfamiliar with node.js at the time and everything was in one large file, and I accepted a PR with this solution.

Its been on my todo list to get some help with the architecture of the whole thing after I got everything working, but never got around to it.  

Feel free to give me some advice :D

Comment (Swizec) [reply utils/getProviders.js]: It's not a bad structure at all. I'm more making excuses for why I couldn't include utils the same way you do elsewhere :)

Tried to keep with the paradigms of your project.

**Comment by raae** [comment]
Closed in favor of  #93

**Pull request closed without merging**